pkg/kubelet/pod_workers.go

```diff
@@ -819,6 +857,44 @@ func (p *podWorkers) acknowledgeTerminating(pod *v1.Pod) PodStatusFunc {
 	return nil
 }
 
+// queueReservedWork queues a reserved work having the same full name as pod
+// and completes the previous pod. It returns true if there is a reserved work
+// else returns false
+func (p *podWorkers) queueReservedWork(pod *v1.Pod) bool {
+	p.podLock.Lock()
+	defer p.podLock.Unlock()
+
+	fullname := kubecontainer.GetPodFullName(pod)
+	work, exists := p.reservedWorkFullnames[fullname]
+	if !exists {
+		return false
+	}
+
+	klog.V(4).InfoS("Queue reserved podWork to sync", "pod", klog.KObj(work.Options.Pod), "podUID", work.Options.Pod.UID)
+	podUpdates := p.podUpdates[pod.UID]
+	p.podUpdates[work.Options.Pod.UID] = podUpdates
+	p.podSyncStatuses[work.Options.Pod.UID].working = true
+	podUpdates <- work
+
+	delete(p.podUpdates, pod.UID)
```

Comment:
Since this is identical to complete terminating, I really didn't expect this to be here (but instead expected the chunk in managePodLoop to invoke this method, then invoke completeTerminating)